### PR TITLE
|Refactoring] In typeMerge(): refactoring returning gotos into return statements

### PR DESCRIPTION
**Pull request by SSoulaimane**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
If you can think of more descriptive names for the added nested functions that would be great.

**Comment by SSoulaimane** [comment]
> If you can think of more descriptive names for the added nested functions that would be great.

I don't know, what are you thinking?

File: src/dmd/dcast.d [hunk 1]
```diff
@@ -2801,10 +2842,10 @@ bool typeMerge(Scope* sc, TOK op, Type* pt, Expression* pe1, Expression* pe2)
         }
     }
 
-    Type t1 = e1.type;
-    Type t2 = e2.type;
```
Comment (thewilsonator) [inline src/dmd/dcast.d]: These are now redundant.

File: src/dmd/dcast.d [hunk 2]
```diff
@@ -2801,10 +2842,10 @@ bool typeMerge(Scope* sc, TOK op, Type* pt, Expression* pe1, Expression* pe2)
...
+    t1 = e1.type;
+    t2 = e2.type;
     assert(t1);
-    Type t = t1;
```
Comment (thewilsonator) [inline src/dmd/dcast.d]: This should be moved to its declaration.

File: src/dmd/dcast.d (hunk 1 above)
Comment (SSoulaimane) [reply src/dmd/dcast.d]: there is this change
https://github.com/dlang/dmd/blob/5b3a28416610535a4d415e0f27323ed64ea3349f/src/dmd/dcast.d#L2840-L2841

File: src/dmd/dcast.d (hunk 2 above)
Comment (SSoulaimane) [reply src/dmd/dcast.d]: same.

**Pull request merged**